ol/smoke-tests/src/tests.rs

```diff
@@ -66,5 +287,168 @@ async fn ol_test_basic_restartability() {
     let client = validator.rest_client();
     swarm.chain_info().ol_send_demo_tx_root(Some(client)).await.expect("could not send tx");
     dbg!("tx sent");
-  
-}
\ No newline at end of file
+}
+
+#[tokio::test]
+async fn ol_test_mining() {
+    let mut test_environment = TestEnvionement::new().await;
+
+    let persona = "alice";
+
+    let mnemonic = test_environment.get_persona_mnemonic(persona);
+    let mnem = Mnemonic::from(&mnemonic).unwrap();
+
+    let mut wallet = test_environment.get_persona_wallet(persona);
+    wallet.generate_addresses(1).unwrap();
+    let address = wallet.get_addresses().unwrap().into_iter().next().unwrap();
+    let private_key = wallet.get_private_key(&address).unwrap();
+    let account_key = AccountKey::from_private_key(private_key);
+    let new_account = LocalAccount::new(address, account_key, 0);
+
+    test_environment.fund_account(&new_account).await;
+
+    let validator = test_environment.swarm.validators().next().unwrap();
+    let rpc_client = validator.json_rpc_client();
+
+    match rpc_client.get_miner_state(address) {
+        Err(err) => {
+            let err = err.json_rpc_error().unwrap();
+            assert_eq!(err.message, "Server error: could not get tower state");
+        },
+        _ => {
+            panic!("miner state for new account shouldn't exists");
+        },
+    }
+
+    // Proof #0
+    let proof_path = test_environment.metadata.workspace_root.join(
+        format!(
```

Comment:
This looks good.
I'll just point out we have some helpers for this kind of thing in:
ol/types/src/fixtures.rs